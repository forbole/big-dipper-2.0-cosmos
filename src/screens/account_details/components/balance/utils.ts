import { formatNumber } from '@utils/format_token';
import Big from 'big.js';

export const formatBalanceData = (data: {
  available: TokenUnit;
  delegate: TokenUnit;
  unbonding: TokenUnit;
  reward: TokenUnit;
  commission?: TokenUnit;
  total?: TokenUnit;
}) => {
  const balanceChart = [
    {
      key: 'balanceAvailable',
      display: `${formatNumber(data.available.value, data.available.exponent)} ${data.available.displayDenom.toUpperCase()}`,
      value: data.available.value,
    },
    {
      key: 'balanceDelegate',
      display: `${formatNumber(data.delegate.value, data.delegate.exponent)} ${data.delegate.displayDenom.toUpperCase()}`,
      value: data.delegate.value,
    },
    {
      key: 'balanceUnbonding',
      display: `${formatNumber(data.unbonding.value, data.unbonding.exponent)} ${data.unbonding.displayDenom.toUpperCase()}`,
      value: data.unbonding.value,
    },
    {
      key: 'balanceReward',
      display: `${formatNumber(data.reward.value, data.reward.exponent)} ${data.reward.displayDenom.toUpperCase()}`,
      value: data.reward.value,
    },
  ];

  if (data.commission && Big(data.commission.value).gt(0)) {
    balanceChart.push({
      key: 'balanceCommission',
      display: `${formatNumber(data.commission.value, data.commission.exponent)} ${data.commission.displayDenom.toUpperCase()}`,
      value: data.commission.value,
    });
  }

  return balanceChart;
};
