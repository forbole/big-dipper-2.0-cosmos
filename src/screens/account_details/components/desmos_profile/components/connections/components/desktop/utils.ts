export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'network',
    width: 25,
  },
  {
    key: 'identifier',
    width: 50,
    align: 'right',
  },
  {
    key: 'creationTime',
    width: 25,
    align: 'right',
  },
];
