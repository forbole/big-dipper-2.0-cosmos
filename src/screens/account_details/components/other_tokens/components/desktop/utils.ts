export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'token',
    width: 25,
  },
  {
    key: 'available',
    width: 25,
    align: 'right',
  },
  {
    key: 'reward',
    width: 25,
    align: 'right',
  },
  {
    key: 'commission',
    width: 25,
    align: 'right',
  },
];
