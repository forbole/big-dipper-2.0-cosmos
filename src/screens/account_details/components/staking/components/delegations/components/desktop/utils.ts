export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'validator',
    width: 25,
  },
  {
    key: 'status',
    width: 15,
  },
  {
    key: 'commission',
    width: 10,
    align: 'right',
  },
  {
    key: 'amount',
    width: 25,
    align: 'right',
  },
  {
    key: 'reward',
    width: 25,
    align: 'right',
  },
];
