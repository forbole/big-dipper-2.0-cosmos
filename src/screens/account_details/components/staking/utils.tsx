import {
  Delegations,
  Redelgations,
  Unbondings,
} from './components';
import {
  RedelegationType, UnbondingType, DelegationType,
} from '../../types';

export const getTabs = (staking: {
  delegations: DelegationType[];
  redelegations: RedelegationType[];
  unbondings: UnbondingType[];
}) => {
  return ([
    {
      id: 0,
      key: 'delegations',
      component: Delegations,
      count: staking.delegations.length,
    },
    {
      id: 1,
      key: 'redelegations',
      component: Redelgations,
      count: staking.redelegations.length,
    },
    {
      id: 2,
      key: 'unbondings',
      component: Unbondings,
      count: staking.unbondings.length,
    },
  ]);
};
