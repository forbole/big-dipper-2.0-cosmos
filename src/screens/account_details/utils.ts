import axios from 'axios';
import * as R from 'ramda';
import { toValidatorAddress } from '@utils/prefix_convert';
import {
  AccountCommissionDocument,
  AccountWithdrawalAddressDocument,
  AccountBalancesDocument,
  AccountDelegationBalanceDocument,
  AccountUnbondingBalanceDocument,
  AccountDelegationRewardsDocument,
} from '@src/graphql/general/account_details_documents';

export const fetchCommission = async (address: string) => {
  const defaultReturnValue = {
    commission: {
      coins: null,
    },
  };
  try {
    const { data } = await axios.post(process.env.NEXT_PUBLIC_GRAPHQL_URL, {
      variables: {
        validatorAddress: toValidatorAddress(address),
      },
      query: AccountCommissionDocument,
    });
    return R.pathOr(defaultReturnValue, ['data'], data);
  } catch (error) {
    return defaultReturnValue;
  }
};

export const fetchAccountWithdrawalAddress = async (address: string) => {
  const defaultReturnValue = {
    withdrawalAddress: {
      address,
    },
  };
  try {
    const { data } = await axios.post(process.env.NEXT_PUBLIC_GRAPHQL_URL, {
      variables: {
        address,
      },
      query: AccountWithdrawalAddressDocument,
    });
    return R.pathOr(defaultReturnValue, ['data'], data);
  } catch (error) {
    return defaultReturnValue;
  }
};

export const fetchAvailableBalances = async (address: string) => {
  const defaultReturnValue = {
    accountBalances: {
      coins: [],
    },
  };
  try {
    const { data } = await axios.post(process.env.NEXT_PUBLIC_GRAPHQL_URL, {
      variables: {
        address,
      },
      query: AccountBalancesDocument,
    });
    return R.pathOr(defaultReturnValue, ['data'], data);
  } catch (error) {
    return defaultReturnValue;
  }
};

export const fetchDelegationBalance = async (address: string) => {
  const defaultReturnValue = {
    delegationBalance: {
      coins: [],
    },
  };
  try {
    const { data } = await axios.post(process.env.NEXT_PUBLIC_GRAPHQL_URL, {
      variables: {
        address,
      },
      query: AccountDelegationBalanceDocument,
    });
    return R.pathOr(defaultReturnValue, ['data'], data);
  } catch (error) {
    return defaultReturnValue;
  }
};

export const fetchUnbondingBalance = async (address: string) => {
  const defaultReturnValue = {
    unbondingBalance: {
      coins: [],
    },
  };
  try {
    const { data } = await axios.post(process.env.NEXT_PUBLIC_GRAPHQL_URL, {
      variables: {
        address,
      },
      query: AccountUnbondingBalanceDocument,
    });
    return R.pathOr(defaultReturnValue, ['data'], data);
  } catch (error) {
    return defaultReturnValue;
  }
};

export const fetchRewards = async (address: string) => {
  const defaultReturnValue = {
    delegationRewards: [],
  };
  try {
    const { data } = await axios.post(process.env.NEXT_PUBLIC_GRAPHQL_URL, {
      variables: {
        address,
      },
      query: AccountDelegationRewardsDocument,
    });
    return R.pathOr(defaultReturnValue, ['data'], data);
  } catch (error) {
    return defaultReturnValue;
  }
};
