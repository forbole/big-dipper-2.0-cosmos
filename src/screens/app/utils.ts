import { generalConfig } from '@configs';

export const OPEN_GRAPH_SEO = {
  type: 'website',
  site_name: 'Big Dipper',
  images: [
    {
      url: generalConfig.previewImage,
      width: 800,
      height: 600,
      alt: 'Preview Photo',
    },
  ],
};

export const TWITTER_SEO = {
  cardType: 'summary_large_image',
};

export const ADDITIONAL_LINK_TAGS_SEO = [
  {
    rel: 'apple-touch-icon',
    href: '/icons/apple-touch-icon.png',
    sizes: '180x180',
  },
  {
    rel: 'icon',
    type: 'image/png',
    href: '/icons/favicon-32x32.png',
    sizes: '32x32',
  },
  {
    rel: 'icon',
    type: 'image/png',
    href: '/icons/favicon-16x16.png',
    sizes: '16x16',
  },
  {
    rel: 'manifest',
    href: '/icons/site.webmanifest',
  },
  {
    rel: 'mask-icon',
    href: '/icons/safari-pinned-tab.svg',
    color: '#5bbad5',
  },
  {
    rel: 'shortcut icon',
    href: '/icons/favicon.ico',
  },
];

export const ADDITIONAL_META_TAGS = [
  {
    property: 'viewport',
    content: 'minimum-scale=1, initial-scale=1, width=device-width',
  },
  {
    property: 'msapplication-TileColor',
    content: '#da532c',
  },
  {
    name: 'msapplication-config',
    content: '/icons/browserconfig.xml',
  },
  {
    name: 'theme-color',
    content: '#ffffff',
  },
];
