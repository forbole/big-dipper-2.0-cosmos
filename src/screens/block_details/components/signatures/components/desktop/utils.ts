export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'validator',
    width: 100,
  },
];
