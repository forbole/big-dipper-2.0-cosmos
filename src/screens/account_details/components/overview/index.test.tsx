import React from 'react';
import renderer from 'react-test-renderer';
import { MockTheme } from '@tests/utils';
import Overview from '.';

// ==================================
// mocks
// ==================================
jest.mock('@components', () => ({
  BoxDetails: (props) => <div id="BoxDetails" {...props} />,
  Box: (props) => <div id="Box" {...props} />,
}));

// ==================================
// unit tests
// ==================================
describe('screen: AccountDetails/Overview', () => {
  it('matches snapshot', () => {
    const component = renderer.create(
      <MockTheme>
        <Overview
          withdrawalAddress="withdrawalAddress"
          address="address"
        />
      </MockTheme>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
