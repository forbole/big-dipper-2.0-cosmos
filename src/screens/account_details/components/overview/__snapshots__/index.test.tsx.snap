// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: AccountDetails/Overview matches snapshot 1`] = `
<div
  className="makeStyles-root"
  id="Box"
>
  <div
    className="makeStyles-copyText makeStyles-item"
  >
    <p
      className="MuiTypography-root label MuiTypography-body1"
    >
      accounts:address
    </p>
    <div
      className="detail"
    >
      <
        className="makeStyles-actionIcons"
        onClick={[Function]}
      />
      <
        className="makeStyles-actionIcons"
        onClick={[Function]}
      />
      <p
        className="MuiTypography-root value MuiTypography-body1"
      >
        address
      </p>
    </div>
  </div>
  <div
    className="makeStyles-copyText makeStyles-item"
  >
    <p
      className="MuiTypography-root label MuiTypography-body1"
    >
      accounts:rewardAddress
    </p>
    <div
      className="detail"
    >
      <
        className="makeStyles-actionIcons"
        onClick={[Function]}
      />
      <p
        className="MuiTypography-root value MuiTypography-body1"
      >
        withdrawalAddress
      </p>
    </div>
  </div>
</div>
`;
